helper/resource/importstate/import_block_with_id_test.go

```diff
@@ -376,3 +379,50 @@ func TestImportBlock_WithID_WithBlankComputedAttribute_GeneratesCorrectPlan(t *t
 		},
 	})
 }
+
+func TestImportBlock_WithID_WithExternalProvider(t *testing.T) {
+	t.Parallel()
+
+	config := `
+resource "random_string" "mystery_message" {
+	length = 31
+}
+`
+
+	configWithImportBlock := config + `
+import {
+	to = random_string.mystery_message
+	id = "It was a dark and stormy night."
+}
+`
+
+	r.UnitTest(t, r.TestCase{
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			tfversion.SkipBelow(tfversion.Version1_5_0), // ImportBlockWithID requires Terraform 1.5.0 or later
+		},
+		ExternalProviders: map[string]r.ExternalProvider{
+			"random": {
+				Source: "hashicorp/random",
+			},
+		},
+		Steps: []r.TestStep{
+			{
+				Config: config,
+			},
+			{
+				ImportState:     true,
+				ImportStateKind: r.ImportBlockWithID,
+				Config:          configWithImportBlock,
```

Comment:
Just to confirm my understanding, provider developers wouldn't need to use a config in this scenario to do this (if the ID needed to be explicitly defined), they could just use `ImportStateId`:

```go
			{
				ImportState:     true,
				ImportStateKind: r.ImportBlockWithID,
				ImportStateId:   "It was a dark and stormy night.",
				ResourceName:    "random_string.mystery_message",
				ImportPlanChecks: r.ImportPlanChecks{
					PreApply: []plancheck.PlanCheck{
						plancheck.ExpectKnownValue(
							"random_string.mystery_message",
							tfjsonpath.New("result"),
							knownvalue.StringExact("It was a dark and stormy night.")),
					},
				},
```

Reply:
> `ImportStateId`

Nice catch. We could come up with a different example here so that it better models a situation that calls for an inline config.